<?xml version="1.0"?> 
<gazebo version="1.0">
  <world name="default">
    <scene>
      <ambient rgba="0.1 0.1 0.1 1"/>
      <background rgba="0 0 0 1.0">
        <sky material="Gazebo/CloudySky"/>
      </background>
      <shadows enabled="true"/>
      <grid enabled="false"/>
    </scene>

    <physics type="ode" update_rate="100">
      <gravity xyz="0 0 -9.8"/>
      <ode>
        <solver type="quick" dt="0.01" iters="10" sor="1.3"/>
        <constraints cfm="0.0" erp="0.2" contact_max_correcting_vel="100.0" contact_surface_layer="0.001"/>
      </ode>
    </physics>

    <model name="plane1_model" static="true">
      <origin pose="174.4 -71 1.99 0 0 0"/>
      <link name="body">

        <collision name="geom">
          <geometry>
            <plane normal="0 0 1"/>
          </geometry>
        </collision>

        <visual name="visual" cast_shadows="false">
          <geometry>
            <plane normal="0 0 1"/>
          </geometry>
          <material script="Gazebo/Grey"/>
        </visual>
      </link>
    </model>

    <model name="box1" static="true">
      <origin pose="178 -74.0 3.0 0 0 0"/>
      <link name="body">
        <inertial mass="0.001">
          <inertia ixx=".001" ixy="0" ixz="0" iyy=".001" iyz="0" izz=".001"/>
        </inertial>
        <collision name="collision">
          <geometry>
            <box size="0.05 0.05 0.15"/>
          </geometry>
        </collision>

        <visual name="visual" cast_shadows="false">
          <geometry>
            <box size="0.05 0.05 0.15"/>
          </geometry>
          <material script="Gazebo/Grey"/>
        </visual>
      </link>
    </model>


    <include filename="quadrotor.model" model_pose="-90 42 30 0 0 0"/>

    <include filename="pr2.model" model_name="pr2"
      model_pose="178 -74.66 2 0 0 1.5707">
      <plugin filename="libmovie_pr2.so" name="pr2_movie"/>
    </include>
    <include filename="youbot_static.model"
             model_pose="178.8 -74.66 2.09 0 0 -1.5707"/>

    <include filename="pioneer2dx.model" model_name="pioneer2dx"
      model_pose="161.0 -92 2 0 0 1.5707">
      <plugin filename="libmovie_pioneer2dx.so" name="pioneer2dx_movie"/>
    </include>

    <include filename="willowgarage.model" model_name="wg"
      model_pose="150 -90 2.0 0 0 0"/>

    <model name="terrain" static="true">
      <link name="link">
        <visual name="visual">
          <geometry>
            <heightmap filename="canyon.png" size="513 513 10"
              offset="0 0 0"/>
          </geometry>
        </visual>
      </link>
    </model>

    <light type="directional" name="sun" cast_shadows="true">
      <origin pose="0 0 500 0 0 0"/>
      <diffuse rgba="1.0 1.0 1.0 1"/>
      <specular rgba="0.2 0.2 0.2 1"/>
      <attenuation range="200" constant="0.8" linear="0.0" quadratic="0.0"/>
      <direction xyz="0.0 0.2 -1.0"/>
    </light>
  </world>
</gazebo>
